extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -696,7 +736,8 @@ class LifecycleTask extends BackbeatTask {
         const daysSinceInitiated = this._findDaysSince(new Date(staleDate));
         const ncve = 'NoncurrentVersionExpiration';
         const ncd = 'NoncurrentDays';
-        const doesNCVExpirationRuleApply = (rules[ncve] && rules[ncve][ncd] &&
+        const doesNCVExpirationRuleApply = (rules[ncve] &&
```

Comment:
Maybe a better fix can be to change the test to `rules[ncve][ncd] !== undefined`? Then it would not require special handling of test case in the main code.

Reply:
nice I like that, will edit :)